diff -ur xmr-stak-cpu-master/minethd.cpp xmr-stak-cpu-master-new/minethd.cpp
--- xmr-stak-cpu-master/minethd.cpp	2017-04-29 12:20:06.000000000 -0400
+++ xmr-stak-cpu-master-new/minethd.cpp	2017-05-03 03:07:08.313778313 -0400
@@ -26,6 +26,7 @@
 #include <chrono>
 #include <thread>
 #include <bitset>
+#include <cstring>
 #include "console.h"
 
 #ifdef _WIN32
diff -ur xmr-stak-cpu-master/socks.h xmr-stak-cpu-master-new/socks.h
--- xmr-stak-cpu-master/socks.h	2017-04-29 12:20:06.000000000 -0400
+++ xmr-stak-cpu-master-new/socks.h	2017-05-03 03:06:20.306575117 -0400
@@ -76,12 +76,8 @@
 inline const char* sock_strerror(char* buf, size_t len)
 {
 	buf[0] = '\0';
-#if defined(__APPLE__)
 	strerror_r(errno, buf, len);
 	return buf;
-#else
-	return strerror_r(errno, buf, len);
-#endif
 }
 
 inline const char* sock_gai_strerror(int err, char* buf, size_t len)
